tests/testthat/test_ggetbout.R

```diff
@@ -1,91 +1,54 @@
-  bm1Eboutcount = c(rep(0,42), rep(1,131), rep(0,11), rep(1,123), rep(0,693))
-  expect_that(sum(bm1$x),equals(228))
-  expect_that(sum(bm1$boutcount),equals(254))
-  expect_that(bm1$x,equals(bm1Ex))
-  expect_that(bm1$boutcount,equals(bm1Eboutcount))
   
-  # Metric 2
-  set.seed(300)
-  bm2 = g.getbout(x=round(runif(1000,0.4,1)),boutduration = 120,boutcriter=0.9,
-                  closedbout=FALSE,bout.metric=2,ws3=5)
-
-  bm2Ex = c(rep(0,42), rep(1,286), rep(0,672))
-  bm2Eboutcount = c(rep(0,42), rep(1,286), rep(0,672))
+  for (ws3 in c(5, 15, 30, 60)) { # epoch size should not influence the result, so we can turn it into a loop
+    cat(paste0("\nepochsize ", ws3))
+    boutcriter = 1
+    N = 60 / ws3 # N is number of epochs in minute
+    boutduration = 10 * N
+    # Metric 7 - 2 bouts out of which 1 is long enough
+    xtest = c(rep(0, 15 * N), rep(1, 6 * N), rep(0, 1 * N), rep(1, 10 * N), rep(0, 20 * N))
+    bouts1 = g.getbout(x = xtest, boutduration = boutduration, boutcriter = boutcriter, ws3 = ws3)
+    expect_equal(sum(bouts1), 10 * N)
+    expect_equal(bouts1, c(rep(0, 22 * N), rep(1, 10 * N), rep(0, 20 * N)))
+    expect_equal(bouts1, c(rep(0, 22 * N), rep(1, 10 * N), rep(0, 20 * N)))
+    
+    # Metric 7 - ignore gaps when boutcriter is 1
+    xtest = x = c(rep(1, 4 * N), rep(0, 3 * N), rep(1, 6 * N),rep(0, 3 * N), rep(1, 10 * N), rep(0, 1 * N) , rep(1, 9 * N))
+    bouts2 = g.getbout(x = xtest, boutduration = boutduration, boutcriter = boutcriter, ws3 = ws3)
+    expect_equal(sum(bouts2), 10 * N)
+    expect_equal(sum(bouts2), 10 * N)
+    
+    # Metric 7 - ability to detect odd length bouts
+    boutduration = 9 * N
+    xtest = x = c(rep(0, 20 * N), rep(1, 9 * N), rep(0, 20 * N))
+    bouts3 = g.getbout(x = xtest, boutduration = boutduration, boutcriter = boutcriter, ws3 = ws3)
+    expect_equal(sum(bouts3), 9 * N)
+    
+    # Metric 7 - ability to detect bouts with grace periods
+    xtest = c(rep(0, 15 * N), rep(1,6 * N), rep(0, 1 * N), rep(1,10 * N), rep(0,20 * N))
+    boutcriter = 0.9
+    boutduration = 10 * N
+    bouts4 = g.getbout(x = xtest, boutduration = boutduration, boutcriter = boutcriter, ws3 = ws3)
+    expect_equal(sum(bouts4), 17 * N)
+    
+    # Metric 7 - ability to detect bouts with grace periods, but also ignore too many grace periods
+    xtest = c(rep(0, 15 * N), rep(1, 6 * N), rep(0, 1 * N), rep(1,3 * N), rep(0, 1 * N), rep(1, 4 * N), rep(0, 20 * N))
+    bouts5 = g.getbout(x = xtest, boutduration = boutduration, boutcriter = boutcriter, ws3 = ws3)
+    expect_equal(sum(bouts5), 10 * N)
+    
+    
+    # Metric 7 - bout length epquals epoch length
+    boutduration = 1
+    xtest = c(rep(0, 10 * N), 1, rep(0, 10 * N))
+    bouts1 = g.getbout(x = xtest, boutduration = boutduration, boutcriter = boutcriter, ws3 = ws3)
+    expect_equal(sum(bouts1), 1)
+    expect_equal(bouts1, xtest)
+    expect_equal(bouts1, xtest)
+  }
```

Comment:
Should we also test that the bout starts/ends at the epochs it is expected? We then would make sure that light PA bouts do not overlap previously detected MVPA or Inactivity bouts. I can add this to the test if you agree.